lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -487,6 +491,7 @@ def reconfig_redirect_opts!(res, opts)
 
       opts['rhost'] = datastore['RHOST']
       opts['vhost'] = opts['vhost'] || opts['rhost'] || self.vhost()
+      opts['ssl_server_name_indication'] = opts['vhost']
```

Comment:
If the location is relative, it's on the same target host meaning we should use the explicit `SSLServerNameIndication` right? Not sure it even matters though if the underlying TCP connection is kept open. :thinking: 

```suggestion
      opts['ssl_server_name_indication'] = datastore['SSLServerNameIndication'] || opts['vhost']
```

Reply:
So I was able to find a case were this didn't work as expected. Using the `auxiliary/scanner/http/title` module and the URL `https://blog.xpnsec.com/tailoring-cobalt-strike-on-target` represents an ideal use case because the host requires SNI and that resource will result in a 302 relative redirect to `https://blog.xpnsec.com/tailoring-cobalt-strike-on-target/`.

When scanning that site (`run https://blog.xpnsec.com/tailoring-cobalt-strike-on-target`), no results are returned.

Changing it to this however fixed it:

```suggestion
      opts['vhost'] = opts['vhost'] || self.vhost || opts['rhost']
      opts['ssl_server_name_indication'] = opts['vhost']
```

Reply:
I was also able to confirm that it looks like a regression. By running: `./msfconsole -x "version; use auxiliary/scanner/http/title; run https://blog.xpnsec.com/tailoring-cobalt-strike-on-target FollowRedirect=true"`

Master branch:
```
Framework: 6.1.42-dev-855d77e763
Console  : 6.1.42-dev-855d77e763
[+] [52.73.153.209:443] [C:301] [R:/tailoring-cobalt-strike-on-target/] [S:Netlify] Tailoring Cobalt Strike on Target - XPN InfoSec Blog
[*] Scanned 1 of 4 hosts (25% complete)
[+] [34.194.149.67:443] [C:301] [R:/tailoring-cobalt-strike-on-target/] [S:Netlify] Tailoring Cobalt Strike on Target - XPN InfoSec Blog
[*] Scanned 1 of 4 hosts (25% complete)
[*] Scanned 2 of 4 hosts (50% complete)
[*] Scanned 3 of 4 hosts (75% complete)
[*] Scanned 4 of 4 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/title) >
```

PR Branch:
```
Framework: 6.1.41-dev-92bdcda1af
Console  : 6.1.41-dev-92bdcda1af
[*] Scanned 1 of 4 hosts (25% complete)
[*] Scanned 2 of 4 hosts (50% complete)
[*] Scanned 3 of 4 hosts (75% complete)
[*] Scanned 4 of 4 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/title) > 
```

Reply:
I applied your suggestion as a second commit, hopefully that ticks the boxes 🤞